### PR TITLE
docs: addition of method documentation

### PR DESCRIPTION
**Pull request by JavierDelgadoFernandez**
* Improves method documentation. (addresses #16)

Signed-off-by: Javier Delgado javier.delgado.fernandez@cern.ch

### CODE REVIEW

**Comment by JavierDelgadoFernandez** [comment]
@hachreak Could you take a look please? :smile:

**Comment by hachreak** [comment]
:+1:

**Comment by hachreak** [comment]
could you add `tasks`, `ext` inside `docs/api.rst`? :)

File: invenio_mail/ext.py [hunk 1]
```diff
@@ -49,23 +49,33 @@ def print_email(message, app):
...
 
-    :param app: Flask application object.
-    :param stream: Stream to print emails to. Defaults to ``sys.stdout``.
-    """
+        Mails are only printed to the stream if ``MAIL_SUPPRESS_SEND`` is
+        ``True``.
 
-    def __init__(self, app=None, stream=None):
-        """Extension initialization."""
+        :param app: Flask application object.
```
Comment (jirikuncar) [inline invenio_mail/ext.py]: @JavierDelgadoFernandez make sure that you include `autoclass_content = 'both'` in `docs/conf.py`.

Comment (JavierDelgadoFernandez) [inline invenio_mail/ext.py]: What do you think about https://github.com/inveniosoftware/invenio-mail/pull/18/files#diff-85987f48f1258d9ee486e3191495582dR328?

File: docs/conf.py [hunk 2]
```diff
@@ -325,6 +325,9 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 #texinfo_no_detailmenu = False
 
+# Which content should be inserted into the main body of the autoclass
```
Comment (jirikuncar) [inline docs/conf.py]: please check https://github.com/inveniosoftware/invenio-i18n/blob/master/docs/conf.py#L334-L335

File: invenio_mail/ext.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_mail/ext.py]: Yes. Please check https://github.com/inveniosoftware/invenio-i18n/blob/master/docs/conf.py#L334-L335

File: docs/api.rst [hunk 3]
```diff
@@ -25,5 +25,12 @@
...
 ========
 
+Flask extension
+---------------
 .. automodule:: invenio_mail.ext
    :members:
+
+Tasks
+---------
+.. automodule:: invenio_mail.tasks
```
Comment (hachreak) [inline docs/api.rst]: it looks like that the documentation for the tasks is not included (only the title of the page).
Maybe it's a problem of celery? (e.g. also the [Celery tasks](https://pythonhosted.org/invenio-indexer/api.html#module-invenio_indexer.tasks) page of invenio-indexer is empty)

**Comment by hachreak** [comment]
apart the small issue with `tasks`, the rest is LGTM.

File: docs/api.rst (hunk 3 above)
Comment (jirikuncar) [inline docs/api.rst]: Please check: http://docs.celeryproject.org/en/latest/reference/celery.contrib.sphinx.html

File: docs/api.rst [hunk 4]
```diff
@@ -25,5 +25,12 @@
 API Docs
 ========
 
+Flask extension
```
Comment (jirikuncar) [inline docs/api.rst]: IMHO this subsection is not necessary here.

File: docs/conf.py [hunk 5]
```diff
@@ -44,6 +44,7 @@
     'sphinx.ext.doctest',
     'sphinx.ext.intersphinx',
     'sphinx.ext.viewcode',
+    'celery.contrib.sphinx',
```
Comment (JavierDelgadoFernandez) [inline docs/conf.py]: @jirikuncar Still not working the task documentation.

File: docs/api.rst [hunk 6]
```diff
@@ -27,3 +27,7 @@ API Docs
 
 .. automodule:: invenio_mail.ext
    :members:
+
+Tasks
+-----
+.. autotask:: invenio_mail.tasks.send_email
```
Comment (JavierDelgadoFernandez) [inline docs/api.rst]: @jirikuncar @hachreak found that in this way it is working.

**Pull request merged**